feat(NIST 800-53): add DynamoDB rules to pack (#97)
cdklabs/cdk-nag: RULES.md

| ------------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | [NIST.800.53-DynamoDBPITREnabled](https://docs.aws.amazon.com/config/latest/developerguide/dynamodb-pitr-enabled.html)             | DynamoDB does not have PITR enabled - (Control IDs: CP-9(b), CP-10, SI-12).                                 | It maintains the backups by ensuring that point-in-time recovery is enabled in Amazon DynamoDB. The recovery maintains continuous backups of your table for the last 35 days.                                                                                                                                                                                                                                           |

Reviewer: ```suggestion
| [NIST.800.53-DynamoDBPITREnabled](https://docs.aws.amazon.com/config/latest/developerguide/dynamodb-pitr-enabled.html)             | 'The DynamoDB table does not have Point-in-time Recovery enabled - (Control IDs: CP-9(b), CP-10, SI-12).                                 | The recovery maintains continuous backups of your table for the last 35 days.                                                                                                                                                                                                                                           |
```

> | [NIST.800.53-EC2CheckDetailedMonitoring](https://docs.aws.amazon.com/config/latest/developerguide/ec2-instance-detailed-monitoring-enabled.html)             | The EC2 instance does not have detailed monitoring enabled - (Control IDs: CA-7(a)(b), SI-4(2), SI-4(a)(b)(c)).                                 | It maintains the backups by ensuring that point-in-time recovery is enabled in Amazon DynamoDB. The recovery maintains continuous backups of your table for the last 35 days.                                                                                                                                                                                                                                           |

Reviewer: I think this was accidentally changed

```suggestion
| [NIST.800.53-EC2CheckDetailedMonitoring](https://docs.aws.amazon.com/config/latest/developerguide/ec2-instance-detailed-monitoring-enabled.html) | The EC2 instance does not have detailed monitoring enabled - (Control IDs: CA-7(a)(b), SI-4(2), SI-4(a)(b)(c)). | Detailed monitoring provides additional monitoring information (such as 1-minute period graphs) on the AWS console. |
```

| AWS Config Rule Name                                                                                                 | Summary                                                                                                 | Relevant Control IDs                 |
| -------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------- | ------------------------------------ |
| [dynamodb-autoscaling-enabled](https://docs.aws.amazon.com/config/latest/developerguide/dynamodb-autoscaling-enabled.html)                   | DynamoDB has autoscaling enabled.                              | CP-10, SC-5                                 |
| [dynamodb-in-backup-plan](https://docs.aws.amazon.com/config/latest/developerguide/dynamodb-in-backup-plan.html)                   | DynamoDB tables are in an AWS Backup Plan.                              | CP-9(b), CP-10, SI-12                                 |

> | [dynamodb-table-encrypted-kms](https://docs.aws.amazon.com/config/latest/developerguide/dynamodb-table-encrypted-kms.html)                   | DynamoDB tables are encrypted in KMS.                              |  SC-13                                 |

Reviewer: I think this rule needs a footnote. It's not implemented because it's unnecessary to implement

```
| [dynamodb-table-encrypted-kms](https://docs.aws.amazon.com/config/latest/developerguide/dynamodb-table-encrypted-kms.html)                   | DynamoDB tables are encrypted in KMS.<sup>[1](#dynamodb-table-encrypted-kms)</sup>                             |  SC-13                                 |
```

And then add a Footnotes section below
```
## Footnotes

<a name="dynamodb-table-encrypted-kms">1</a>: This rule is intentionally unimplemented. Server-side encryption at rest is enabled on all DynamoDB table data and [cannot be disabled](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/encryption.usagenotes.html#encryption.usagenotes.tabledata).
```
